eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/ImpTest.hs

```diff
@@ -0,0 +1,630 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Test.Cardano.Ledger.Conway.ImpTest (
```

Comment:
I feel this is obvious, yet I don't get where  the prefix `Imp` comes from :grimacing:  "Implementation" ?

Reply:
Oh nevermind, I see @lehins asked the same below, I will follow that conversation

Reply:
Maybe it stands for "Imp"ortant tests? or maybe those are tests that are "Imp"ossible to fail??

The suspense is killing me! What does it mean? :rofl:

Reply:
Or "Imp"roved? Yes, tell us @Soupstraw :grimacing: :grimacing:

Reply:
imp

noun
1. A mischievous child.
2. A small demon.
3. A graft.

:orange_heart: More at Wordnik

Reply:
@Soupstraw Apparently `Proc` for "procedural" is a befitting word if you were going for `imerative`... like proc_macros from Rust, may be. :slightly_smiling_face:

Reply:
Did you know that `proc` is actually a keyword in Haskell :smile: 
https://wiki.haskell.org/Keywords#proc

Regardless of this fact proc would be a bad suffix.